ring/ring.go

```diff
@@ -125,7 +125,10 @@ func (r *Ring) Get(key uint32, n int, heartbeatTimeout time.Duration) ([]Ingeste
 			continue
 		}
 		distinctHosts[host] = struct{}{}
+	}
 
+	ingesters := make([]IngesterDesc, 0, n)
+	for host := range distinctHosts {
```

Comment:
There's no semantic difference. It's largely to start a conversation with @tomwilkie. Here's the context:

Yesterday afternoon, he said that this loop was wrong because it would go past the bounds of valid ingesters, and that the right thing to do was a two-pass filter. The first finding all the ingesters we _could_ write to and the second selecting the ones we _should_ write to.

At first glance I thought he was right, went to fix it and then noticed that this loop does _not_ go past the bounds of valid ingesters (I alluded to this in the PR description).

Then I figured if Tom read it incorrectly and it took me twenty minutes of staring at code to figure it out, maybe it would be better to split this into two obvious passes.